tvb_framework/tvb/adapters/analyzers/bct_adapters.py

```diff
@@ -130,34 +108,25 @@ def get_required_disk_size(self, view_model):
     def get_connectivity(self, view_model):
         return self.load_traited_by_gid(view_model.connectivity)
 
-    def execute_matlab(self, matlab_code, data):
-        self.matlab_worker.add_to_path(BCT_PATH)
-        self.log.info("Starting execution of MATLAB code:" + matlab_code)
-        runcode, matlablog, result = self.matlab_worker.matlab(matlab_code, data=data, work_dir=self.get_storage_path())
-        self.log.debug("Code run in MATLAB: " + str(runcode))
-        self.log.debug("MATLAB log: " + str(matlablog))
-        self.log.debug("Finished MATLAB execution:" + str(result))
-        return result
-
-    def build_connectivity_measure(self, result, key, connectivity, title="", label_x="", label_y=""):
+    def build_connectivity_measure(self, array_data, connectivity, title="", label_x="", label_y=""):
         measure = ConnectivityMeasure()
-        measure.array_data = result[key]
+        measure.array_data = array_data
         measure.connectivity = connectivity
         measure.title = title
         measure.label_x = label_x
         measure.label_y = label_y
         return self.store_complete(measure)
 
-    def build_float_value_wrapper(self, result, key, title=""):
+    def build_float_value_wrapper(self, result_value, title=""):
         value = ValueWrapper()
-        value.data_value = str(float(result[key]))
+        value.data_value = str(float(result_value))
         value.data_type = 'float'
         value.data_name = title
         return self.store_complete(value)
 
-    def build_int_value_wrapper(self, result, key, title=""):
+    def build_int_value_wrapper(self, result_int, title=""):
         value = ValueWrapper()
-        value.data_value = str(int(result[key]))
+        value.data_value = str(int(result_int))
```

Comment:
if the result is expected to be int is this conversion to int still required?

Reply:
The result is int-like, but not always int, thus the conversion is required.

Reply:
That makes sense. Thanks for the clarification!